app/models/like.rb

```diff
@@ -1,4 +1,11 @@
 class Like < ApplicationRecord
-  belongs_to :author
+  belongs_to :author, class_name: 'User'
   belongs_to :post
+
+  after_create :update_post_likes_counter
+  after_destroy :update_post_likes_counter
+
+  def update_post_likes_counter
+    post.update(likes_counter: post.likes.count)
+  end
```

Comment:
- [ Optional] The same suggestion here, you can use `after_save :update_likes_counter` as well.